src/serializer.ts

```diff
@@ -17,14 +19,157 @@ interface PaymentJSON {
  * Provides functionality to serialize from protocol buffers to JSON objects.
  */
 class Serializer {
+/**
+   * Convert a Transaction to a JSON representation.
+   *
+   * @param {proto.Transaction} transaction A Transaction to convert.
+   * @returns {Object} The Transaction as JSON.
+   */
+  public static transactionToJSON(
+    transaction: Transaction
+  ): object | undefined {
+    // Serialize the protocol buffer to a JSON representation.
+    var object: any = transaction.toObject();
+
+    // Convert fields names where direct conversion is possible.
+    this.convertPropertyName("sequence", "Sequence", object);
+    this.convertPropertyName("signingPublicKey", "SigningPubKey", object);
+    this.convertPropertyName(
+      "lastLedgerSequence",
+      "LastLedgerSequence",
+      object
+    );
+
+    // Delete unsupported fields from the protocol buffer.
+    delete object.memosList;
+    delete object.flags;
+    delete object.signature;
+    delete object.signersList;
+    delete object.sourceTag;
+    delete object.accountTransactionId;
+
+    // Encode SigningPubKey to hex, which is what ripple-binary-codec expects.
+    object.SigningPubKey = Utils.toHex(transaction.getSigningPublicKey_asU8());
+
+    // Convert account field, handling X-Addresses if needed.
+    const accountAddress = transaction.getAccount();
+    if (!accountAddress) {
+      return;
+    }
+    const account = accountAddress.getAddress();
+    if (!account || !Utils.isValidAddress(account)) {
+      return;
+    }
+
+    var normalizedAccount = account;
+    if (Utils.isValidXAddress(account)) {
+      const decodedClassicAddress = Utils.decodeXAddress(account);
+      if (!decodedClassicAddress) {
+        return;
+      }
+
+      // Accounts cannot have a tag.
+      if (decodedClassicAddress.tag !== undefined) {
+        return;
+      }
+
+      normalizedAccount = decodedClassicAddress.address;
+    }
+    object["Account"] = normalizedAccount;
+    delete object.account;
+
+    // Convert XRP denominated fee field.
+    const txFee = transaction.getFee();
+    if (txFee === undefined) {
```

Comment:
I assume Typescript asserts it is either `undefined | number`?

Reply:
Correct.